### PR TITLE
chore(suite-native): homescreen crossroads

### LINKED ISSUE #9649: Create dashboard empty state
Create an empty state that will be displayed inside Homescreen of dashboard. 

https://www.figma.com/file/IidkuNfa5OWtPgta9jF8Gl/T-%E2%80%A2-Handoff?type=design&node-id=2301%3A9624&mode=design&t=7rlDOwZpZ0DvtAuN-1

### PR DESCRIPTION
**Pull request by juriczech**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/9649

## Screenshots:
image

### CODE REVIEW

File: suite-common/wallet-core/src/accounts/accountsReducer.ts [hunk 1]
```diff
@@ -288,3 +285,15 @@ export const selectIsAccountWithRatesByKey = (
 
     return !!rates.find(rate => rate.symbol === account.symbol);
 };
+
+export const selectIsAccountsEmpty = (state: AccountsRootState) =>
+    pipe(selectAccounts(state), A.isEmpty);
+
+export const selectIsPortfolioEmpty = (
```
Comment (Nodonisko) [inline suite-common/wallet-core/src/accounts/accountsReducer.ts:292]: Portfolio is mobile only stuff I think we should keep it in mobile package

File: suite-native/module-accounts-import/src/components/AccountImportSubHeader.tsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ type NavigationProp = StackToTabCompositeProps<
 
 export const AccountImportSubHeader = () => {
     const navigation = useNavigation<NavigationProp>();
-    const userHasAccounts = useSelector(selectUserHasAccounts);
+    const areAccountsEmpty = useSelector(selectIsAccountsEmpty);
```
Comment (vytick) [inline suite-native/module-accounts-import/src/components/AccountImportSubHeader.tsx]: Can we keep either one or the other name?
`are` vs `is`

File: suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    paddingBottom: utils.spacings.extraLarge,
+}));
+
+type NavigationProp = StackToTabCompositeNavigationProp<
+    HomeStackParamList,
+    HomeStackRoutes.Home,
+    RootStackParamList
+>;
+
+export const EmptyAccountsHomescreen = () => {
```
Comment (vytick) [inline suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx]: nit: EmptyAccountsHomescreen -> EmptyAccountsHome**S**creen to be in line with other component names

File: suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    const { applyStyle } = useNativeStyles();
+    const navigation = useNavigation<NavigationProp>();
+
+    const { translate } = useTranslate();
+
+    const colorScheme = useActiveColorScheme();
+
+    const getImageSource = () => {
```
Comment (vytick) [inline suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx]: I suggest to useMemo

File: suite-native/module-home/src/components/EmptyDeviceHomescreen.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+const cardStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginBottom: utils.spacings.extraLarge,
+    padding: utils.spacings.extraLarge,
+}));
+
+export const EmptyDeviceHomescreen = () => {
```
Comment (vytick) [inline suite-native/module-home/src/components/EmptyDeviceHomescreen.tsx]: maybe EmptyDeviceHomeScreen ?

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import { VStack, Card, Button, Text, Pictogram } from '@suite-native/atoms';
+import { useNativeStyles, prepareNativeStyle } from '@trezor/styles';
+import { useTranslate } from '@suite-native/intl';
+import { useDeviceConnect } from '@suite-native/device';
+
+import { SettingsButtonLink } from './SettingsButtonLink';
+
+const cardStyle = prepareNativeStyle<{ flex: 1 | 2 }>((utils, { flex }) => ({
```
Comment (vytick) [inline suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx:19]: maybe to `flex: 1 | 2` this as props type?

File: suite-native/module-onboarding/src/screens/AnalyticsConsentScreen.tsx [hunk 7]
```diff
@@ -130,65 +119,7 @@ export const AnalyticsConsentScreen = () => {
...
-                                                ? chunks[0]
-                                                : ''
-                                        }
-                                    />
-                                ),
-                            }}
-                        />
-                    </Text>
-                </VStack>
+                <AnalyticsConsentContent isEnabled={isEnabled} setIsEnabled={setIsEnabled} />
```
Comment (vytick) [inline suite-native/module-onboarding/src/screens/AnalyticsConsentScreen.tsx]: I see what you did here 😄  👍

**Review by vytick — CHANGES_REQUESTED** [review]
I have a few small comments, but nice job here, again! 😎

File: suite-native/module-onboarding/src/screens/AnalyticsConsentScreen.tsx (hunk 7 above)
Comment (juriczech) [reply suite-native/module-onboarding/src/screens/AnalyticsConsentScreen.tsx]: Looks better IMO. Had to change some stuff here and couldn't resist 😁 Hope you agree

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx (hunk 6 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx:19]: I don't understand your comment. Could you elaborate please?

File: suite-native/module-home/src/components/EmptyDeviceHomescreen.tsx (hunk 5 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyDeviceHomescreen.tsx]: Good point with the cases. However now that I think of it, this is not screen at all and shouldn't have the screen suffix at all. I'll change this a little bit..

File: suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx (hunk 4 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx]: Why? This is computationaly not a heavy operation and I think it's not needed there. But let's discuss, maybe there's valid reason I'm missing. 

Also this is just copy-paste of a component that was deleted from last onboarding so this has already been agreed upon when first onboarding was implemented.

File: suite-native/module-accounts-import/src/components/AccountImportSubHeader.tsx (hunk 2 above)
Comment (juriczech) [reply suite-native/module-accounts-import/src/components/AccountImportSubHeader.tsx:27]: isAccountsEmpty is gramatically incorrect and `are` prefix clearly hints boolean value. What's the problem with plural?

Comment (juriczech) [reply suite-native/module-accounts-import/src/components/AccountImportSubHeader.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/ca1e5d279f0d4e63b5f6c7d5f9006be10fc51f1e

File: suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx (hunk 3 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/4a13f2664427e509f5912f969e64a36545a26bbe

File: suite-native/module-home/src/components/EmptyDeviceHomescreen.tsx (hunk 5 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyDeviceHomescreen.tsx]: What do you think of this?
https://github.com/trezor/trezor-suite/pull/9694/commits/4a13f2664427e509f5912f969e64a36545a26bbe

File: suite-native/app/ios/Podfile.lock [hunk 8]
```diff
@@ -871,4 +871,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: d37876aaf00e7f8309b694188119b139b8452ca1
 
-COCOAPODS: 1.13.0
+COCOAPODS: 1.12.1
```
Comment (PeKne) [inline suite-native/app/ios/Podfile.lock]: downgrade?

File: suite-common/wallet-core/src/device/deviceThunks.ts [hunk 9]
```diff
@@ -418,3 +423,14 @@ export const initDevices = createThunk(
         }
     },
 );
+
+export const createImportedAccountThunk = createThunk(
```
Comment (PeKne) [inline suite-common/wallet-core/src/device/deviceThunks.ts]: IMO this should be moved to `suite-native/device`, since this thunk does not make sense to use in desktop app.

File: suite-native/app/src/navigation/RootStackNavigator.tsx [hunk 10]
```diff
@@ -27,15 +26,13 @@ const RootStack = createNativeStackNavigator<RootStackParamList>();
...
 export const RootStackNavigator = () => {
     const isOnboardingFinished = useSelector(selectIsOnboardingFinished);
-    const userHasAccounts = useSelector(selectUserHasAccounts);
     const { isUsbDeviceConnectFeatureEnabled } = useIsUsbDeviceConnectFeatureEnabled();
 
     const getInitialRouteName = () => {
         if (isOnboardingFinished) {
-            if (userHasAccounts) return RootStackRoutes.AppTabs;
-            if (isUsbDeviceConnectFeatureEnabled) return RootStackRoutes.ConnectDevice;
+            if (isUsbDeviceConnectFeatureEnabled) return RootStackRoutes.AppTabs;
```
Comment (PeKne) [inline suite-native/app/src/navigation/RootStackNavigator.tsx]: This doesn't seem to be correct. This if resolves into AppTabs route, but right underneath is the AppTabs route returned anyway.

File: suite-common/wallet-core/src/accounts/accountsReducer.ts [hunk 11]
```diff
@@ -288,3 +285,15 @@ export const selectIsAccountWithRatesByKey = (
 
     return !!rates.find(rate => rate.symbol === account.symbol);
 };
+
+export const selectIsAccountsListEmpty = (state: AccountsRootState) =>
```
Comment (PeKne) [inline suite-common/wallet-core/src/accounts/accountsReducer.ts:289]: Since is this selector used in `DiscoveryAssetsLoader` it should select only accounts of the currently active device. Otherwise, it will be almost always `true` if you have more than one device.

File: suite-common/wallet-core/src/accounts/accountsReducer.ts [hunk 12]
```diff
@@ -288,3 +285,15 @@ export const selectIsAccountWithRatesByKey = (
...
+    pipe(selectAccounts(state), A.isEmpty);
+
+export const selectIsPortfolioEmpty = (
+    state: AccountsRootState & DeviceRootState & DiscoveryRootState,
+) => {
+    const isAccountsListEmpty = selectIsAccountsListEmpty(state);
```
Comment (PeKne) [inline suite-common/wallet-core/src/accounts/accountsReducer.ts:295]: Same here, it should select only accounts that are connected to the hidden fake device.

File: suite-common/wallet-core/src/accounts/accountsReducer.ts [hunk 13]
```diff
@@ -288,3 +285,15 @@ export const selectIsAccountWithRatesByKey = (
 
     return !!rates.find(rate => rate.symbol === account.symbol);
 };
+
+export const selectIsAccountsListEmpty = (state: AccountsRootState) =>
+    pipe(selectAccounts(state), A.isEmpty);
+
+export const selectIsPortfolioEmpty = (
```
Comment (PeKne) [inline suite-common/wallet-core/src/accounts/accountsReducer.ts:292]: Why did you move this from `suite-native` to `suite-common`?

File: suite-native/module-home/src/components/EmptyAccountsState.tsx [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+    paddingBottom: utils.spacings.extraLarge,
+}));
+
+type NavigationProp = StackToTabCompositeNavigationProp<
+    HomeStackParamList,
+    HomeStackRoutes.Home,
+    RootStackParamList
+>;
+
+export const EmptyAccountsState = () => {
```
Comment (PeKne) [inline suite-native/module-home/src/components/EmptyAccountsState.tsx]: I would preffer to name this component so it is clearer it relates to the portfolio tracker empty state (e.g. `EmptyPortfolioTrackerState`). `Accounts` sound too general to me, since the device flow also works with accounts. 

Or if you don't want to mention portfolio tracker maybe `EmptyImportedAccountsState`.

File: suite-native/module-home/src/components/EmptyAccountsState.tsx [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+            <Card style={applyStyle(cardStyle)}>
+                <VStack spacing="large" alignItems="center">
+                    <AlertBox title={translate('moduleHome.emptyState.accounts.alert')} />
+                    <Image
+                        source={getImageSource()}
+                        resizeMode="contain"
+                        style={applyStyle(imageStyle)}
+                    />
+                    <Text variant="titleSmall">
+                        <Translation id="moduleHome.emptyState.accounts.title" />
```
Comment (PeKne) [inline suite-native/module-home/src/components/EmptyAccountsState.tsx]: Same applies for the `accounts` in the translations file. If you gonna rename this file, rename it there as well please.

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx (hunk 6 above)
Comment (vytick) [reply suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx:19]: Sorry. I think making it into a type

```
export type CardStyleProps = {
         flex: 1 | 2
};

const cardStyle = prepareNativeStyle<CardStyleProps> ...
```

File: suite-native/module-onboarding/src/screens/AnalyticsConsentScreen.tsx (hunk 7 above)
Comment (vytick) [reply suite-native/module-onboarding/src/screens/AnalyticsConsentScreen.tsx]: I dont disagree :). As you said personal preferences are various and this is your code :)

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+
+    return (
+        <VStack spacing="medium" flex={1}>
+            <Card style={applyStyle(cardStyle, { flex: 2 })}>
+                <VStack spacing="large" justifyContent="center" alignItems="center">
+                    <Pictogram
+                        icon="trezor"
+                        variant="green"
+                        title={translate(
+                            'moduleConnectDevice.connectCrossroadsScreen.gotMyTrezor.title',
```
Comment (PeKne) [inline suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx]: Since it was moved from `module-connect-device` to `module-home` I believe it should be moved also in the translations file

File: suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx (hunk 4 above)
Comment (vytick) [reply suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx]: I am just following here what Dan has asked in my previous PRs, where he asked using useMemo for this (and I also copy pasted the code there).

https://github.com/trezor/trezor-suite/pull/9651#discussion_r1359467037

File: suite-native/navigation/src/components/TabBar.tsx [hunk 17]
```diff
@@ -31,6 +35,14 @@ const tabBarStyle = prepareNativeStyle<{
...
     const { applyStyle } = useNativeStyles();
     const insets = useSafeAreaInsets();
+    const { isUsbDeviceConnectFeatureEnabled } = useIsUsbDeviceConnectFeatureEnabled();
+    const isDeviceImported = useSelector(selectIsSelectedDeviceImported);
+
+    const isPortfolioEmpty = useSelector(selectIsPortfolioEmpty);
+
+    const isConnectedDevice = isUsbDeviceConnectFeatureEnabled && !isDeviceImported;
+
+    if (isPortfolioEmpty && !isConnectedDevice) return null;
```
Comment (PeKne) [inline suite-native/navigation/src/components/TabBar.tsx:45]: I think this logic should be rather in the `module-home`. Could not we hide the tabBar from there with something like this?  https://stackoverflow.com/a/72658508

File: suite-common/wallet-core/src/device/deviceReducer.ts [hunk 18]
```diff
@@ -665,3 +658,13 @@ export const selectIsDeviceAuthenticityFulfilled = (
         (deviceAuthenticity.error === 'CA_PUBKEY_NOT_FOUND' && deviceAuthenticity.configExpired) // CA_PUBKEY_NOT_FOUND with configExpired is temporary allowed and just logged to Sentry
     );
 };
+
+export const selectIsSelectedDeviceImported = (state: DeviceRootState) => {
+    const device = selectDevice(state);
+    return device?.id === HIDDEN_DEVICE_ID;
+};
+
+export const selectIsSelectedDeviceAuthorized = (state: DeviceRootState) => {
```
Comment (PeKne) [inline suite-common/wallet-core/src/device/deviceReducer.ts]: We already have very similar selector `selectIsDeviceConnectedAndAuthorized` Can not you use that one? If they are actually different, then is the naming confusing.

**Review by PeKne — APPROVED** [review]

**Review by PeKne — CHANGES_REQUESTED** [review]
The previous approve was a miss click 😅

File: suite-common/wallet-core/src/device/deviceReducer.ts [hunk 19]
```diff
@@ -11,6 +11,7 @@ import { deviceAuthenticityActions } from '@suite-common/device-authenticity';
 
 import { deviceActions } from './deviceActions';
 import { DiscoveryRootState, selectDiscoveryByDeviceState } from '../discovery/discoveryReducer';
+import { HIDDEN_DEVICE_ID } from './deviceConstants';
```
Comment (Nodonisko) [inline suite-common/wallet-core/src/device/deviceReducer.ts]: I think we should rename this `HIDDEN_DEVICE_ID` to something that will reflect it's for portfolio only like `PORTFOLIO_DEVICE_ID` or like that, because just `HIDDEN` gives you zero context what it is good for

File: suite-common/wallet-core/src/device/deviceThunks.ts (hunk 9 above)
Comment (Nodonisko) [inline suite-common/wallet-core/src/device/deviceThunks.ts]: This doesn't seems to create any account but only device, please rename it to reflect that.

File: suite-native/device/src/hooks/useDeviceConnect.tsx [hunk 20]
```diff
@@ -79,7 +82,7 @@ export const useDeviceConnect = () => {
                 ),
             });
         }
-    }, [isConnectedDeviceUninitialized, showAlert, translate]);
+    }, [isConnectedDeviceUninitialized, showAlert, translate, selectedDevice]);
```
Comment (Nodonisko) [inline suite-native/device/src/hooks/useDeviceConnect.tsx]: this seems to be unused in effect

File: suite-native/module-accounts-import/src/accountsImportThunks.ts [hunk 21]
```diff
@@ -36,13 +32,8 @@ const getAccountTypeFromDescriptor = (
 export const importAccountThunk = createThunk(
     `${actionPrefix}/importAccountThunk`,
     ({ accountInfo, accountLabel, coin }: ImportAssetThunkPayload, { dispatch, getState }) => {
-        const device = selectDeviceById(getState(), HIDDEN_DEVICE_ID);
         const deviceState = HIDDEN_DEVICE_STATE;
 
-        if (!device) {
```
Comment (Nodonisko) [inline suite-native/module-accounts-import/src/accountsImportThunks.ts:42]: Q: Wouldn't be "safer" to keep check here? I am not sure just asking for friend

File: suite-common/wallet-core/src/accounts/accountsReducer.ts (hunk 13 above)
Comment (juriczech) [reply suite-common/wallet-core/src/accounts/accountsReducer.ts:292]: I moved it to accountsReducer since it is selector picking state from accounts. Makes sense to be here IMO. Both device and accounts are in common and it can later be used in desktop as well if needed.

File: suite-native/module-accounts-management/src/components/AccountSettingsRemoveCoinButton.tsx [hunk 22]
```diff
@@ -20,13 +14,18 @@ import {
 } from '@suite-native/navigation';
 import { useAlert } from '@suite-native/alerts';
 
-export const AccountSettingsRemoveCoinButton = ({ accountKey }: { accountKey: AccountKey }) => {
+type AccountSettingsRemoveCoinButtonProps = {
+    accountKey: AccountKey;
```
Comment (Nodonisko) [inline suite-native/module-accounts-management/src/components/AccountSettingsRemoveCoinButton.tsx:18]: Q: Does `AccountKey` reflects also device? Because we can have same account imported and also remembered from some device

File: suite-native/module-home/src/components/EmptyAccountsState.tsx [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+
+export const EmptyAccountsState = () => {
+    const { applyStyle } = useNativeStyles();
+    const navigation = useNavigation<NavigationProp>();
+
+    const { translate } = useTranslate();
+
+    const colorScheme = useActiveColorScheme();
+
+    const getImageSource = () => {
```
Comment (Nodonisko) [inline suite-native/module-home/src/components/EmptyAccountsState.tsx]: I would prefer useMemo here than function, no need to create new function in every rerender and also kind of nicer than function since it's basically static value

File: suite-common/wallet-core/src/device/deviceThunks.ts (hunk 9 above)
Comment (juriczech) [reply suite-common/wallet-core/src/device/deviceThunks.ts]: Desktop will be implementing portfolio tracker in the future as well. It makes sense to keep this logic here I think. There are thunks in common that we don't use either. Should they be in desktop only then? The point of common is to have logic that can be used between BOTH packages. Only because it's not used now doesn't mean it can't be used later. To me, this makes sensse and is not platform specific.

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+                            'moduleConnectDevice.connectCrossroadsScreen.gotMyTrezor.connectButton',
+                        )}
+                    </Button>
+                </VStack>
+            </Card>
+            <Card style={applyStyle(cardStyle, { flex: 1 })}>
+                <VStack spacing="large" justifyContent="center" alignItems="center">
+                    <VStack spacing="small" alignItems="center">
+                        <Text variant="titleSmall" textAlign="center">
+                            {translate(
```
Comment (Nodonisko) [inline suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx]: Prefer using Translation component if you don't need string specifically because you can easily add Rich Text Formatting later if needed

File: suite-native/module-onboarding/src/components/AnalyticsConsentContent.tsx [hunk 25]
```diff
@@ -0,0 +1,103 @@
...
+    maxHeight: 64,
+    paddingHorizontal: utils.spacings.medium,
+    paddingVertical: utils.spacings.small,
+    marginBottom: utils.spacings.small,
+    borderRadius: utils.spacings.medium,
+    backgroundColor: utils.colors.backgroundTertiaryDefaultOnElevation1,
+    width: '100%',
+}));
+
+export const AnalyticsConsentContent = ({
```
Comment (Nodonisko) [inline suite-native/module-onboarding/src/components/AnalyticsConsentContent.tsx]: What's the reason for making component of this? I prefer keeping things together if there is no intention to reuse them.

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-native/module-onboarding/src/components/AnalyticsConsentContent.tsx (hunk 25 above)
Comment (juriczech) [reply suite-native/module-onboarding/src/components/AnalyticsConsentContent.tsx]: Just makes it nicer to read the component above and removes overhead. To me, even if it's not going to be reused, I prefer to separate some parts of code. If we only separated parts of code that is reusable, then some components would get really large. There is no performance or readability downside so when I start to feel like component is getting big, for dev experience, I like to separate it.

File: suite-native/module-accounts-import/src/accountsImportThunks.ts (hunk 21 above)
Comment (juriczech) [reply suite-native/module-accounts-import/src/accountsImportThunks.ts:42]: Hmmm 🤔 The device gets created with app initialization so we are certain the device will be there. And we don't actually need the device here, we just need to create the account with `deviceState`. That, as far as I understand is a unique hash of devices so when we have this, even if we created that device later when we need it, it would map onto the device when selecting it's accounts. If that makes sense? 

Basically here we don't need the device, but we're sure it's already created.

File: suite-common/wallet-core/src/device/deviceReducer.ts (hunk 19 above)
Comment (juriczech) [reply suite-common/wallet-core/src/device/deviceReducer.ts]: `PORTFOLIO_TRACKER_DEVICE_ID` sounds good? Portfolio can be for connected device as well

File: suite-common/wallet-core/src/accounts/accountsReducer.ts (hunk 1 above)
Comment (juriczech) [reply suite-common/wallet-core/src/accounts/accountsReducer.ts:292]: If desktop team will implement the portfolio tracker in future (which as far as i know is the goal), then this selector will be needed so makes sense to have here, no?

File: suite-native/navigation/src/components/TabBar.tsx (hunk 17 above)
Comment (juriczech) [reply suite-native/navigation/src/components/TabBar.tsx:45]: The `module-home` shouldn't control tab bar, `TabBar.tsx` should. Why do you think it's better?

File: suite-common/wallet-core/src/accounts/accountsReducer.ts (hunk 11 above)
Comment (juriczech) [reply suite-common/wallet-core/src/accounts/accountsReducer.ts:289]: I know, but since it was like this before this PR and you're still not allowed to switch between them, we should handle this separately and not bloat this PR. As empty state, this works and we will need to do this but not right now, I would like to do it as part of device manager / switcher. 

This was wrong before this PR as well, so ignore it please.

File: suite-common/wallet-core/src/accounts/accountsReducer.ts (hunk 12 above)
Comment (juriczech) [reply suite-common/wallet-core/src/accounts/accountsReducer.ts:295]: Out of scope for this issue

File: suite-native/module-accounts-management/src/components/AccountSettingsRemoveCoinButton.tsx (hunk 22 above)
Comment (juriczech) [reply suite-native/module-accounts-management/src/components/AccountSettingsRemoveCoinButton.tsx:18]: Not account key AFAIK. But account has a deviceState property which we can use for mapping. 

However if I remember correctly, this was already discussed within the team and decided that this WILL be possible and is okay to have account imported as well as within device, since you will never see them side by side since they're in a different portfolio.

Comment (juriczech) [reply suite-native/module-accounts-management/src/components/AccountSettingsRemoveCoinButton.tsx:18]: Okay, turns out account key has a deviceState inside the key, so we could possibly use that, if needed.

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx (hunk 24 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-native/module-home/src/components/EmptyAccountsState.tsx (hunk 23 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyAccountsState.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-native/device/src/hooks/useDeviceConnect.tsx (hunk 20 above)
Comment (juriczech) [reply suite-native/device/src/hooks/useDeviceConnect.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-common/wallet-core/src/device/deviceThunks.ts (hunk 9 above)
Comment (juriczech) [reply suite-common/wallet-core/src/device/deviceThunks.ts]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-common/wallet-core/src/device/deviceReducer.ts (hunk 19 above)
Comment (juriczech) [reply suite-common/wallet-core/src/device/deviceReducer.ts]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-common/wallet-core/src/device/deviceReducer.ts (hunk 18 above)
Comment (juriczech) [reply suite-common/wallet-core/src/device/deviceReducer.ts]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx (hunk 16 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-native/module-home/src/components/EmptyAccountsState.tsx (hunk 15 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyAccountsState.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-native/module-home/src/components/EmptyAccountsState.tsx (hunk 14 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyAccountsState.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-native/app/src/navigation/RootStackNavigator.tsx (hunk 10 above)
Comment (juriczech) [reply suite-native/app/src/navigation/RootStackNavigator.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

Comment (juriczech) [reply suite-native/app/src/navigation/RootStackNavigator.tsx]: It's correct, just redundant if. Removed

File: suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx (hunk 4 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyAccountsHomescreen.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/c9f9b0587b4ba3bea0645ae010e508e0e0911fc4

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx (hunk 6 above)
Comment (juriczech) [reply suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx:19]: If it's just one prop, I prefer to just do it inline. With more props, what you're suggesting makes sense.

File: suite-native/module-onboarding/src/components/AnalyticsConsentContent.tsx (hunk 25 above)
Comment (juriczech) [reply suite-native/module-onboarding/src/components/AnalyticsConsentContent.tsx]: https://github.com/trezor/trezor-suite/pull/9694/commits/ffb98d75aa1444a4e79281a5c5c374257f53f238

File: suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx (hunk 6 above)
Comment (vytick) [reply suite-native/module-home/src/components/EmptyPortfolioCrossroads.tsx:19]: ok. I am ok with that

**Review by Nodonisko — APPROVED** [review]

File: suite-native/app/ios/Podfile.lock (hunk 8 above)
Comment (juriczech) [reply suite-native/app/ios/Podfile.lock]: Cocoapods is a local dependency so don't worry about it in PR's please. I updated mine so now it's fine but if you want to worry about it in PR's we should enforce version of cocoapods to be used which can be done
https://github.com/trezor/trezor-suite/pull/9694/commits/0347bcf7e7df5749119a6bd8389afa63fe5c71ec

Comment (PeKne) [reply suite-native/app/ios/Podfile.lock]: IMO this file should be never pushed at all if the commit does not introduce a new installed pod. Otherwise it is unrelated change.  Thanks for resolving this.

File: suite-native/navigation/src/components/TabBar.tsx (hunk 17 above)
Comment (PeKne) [reply suite-native/navigation/src/components/TabBar.tsx:45]: Because you need to import unnecessary packages to the `suite-native/navigation` module. It works in this scenarion, but since the `navigation` module is imported in almost every package we have, it is very inclined to cyclical dependency.  For this reason i think it is bad and non-reusable design.


I still believe the hiding of the tab bar should be handled in the specific screen source file.

Comment (juriczech) [reply suite-native/navigation/src/components/TabBar.tsx:45]: Let's open this to others opinions as well @Nodonisko @vytick

Comment (PeKne) [reply suite-native/navigation/src/components/TabBar.tsx:45]: I will have to solve this also in receive flow, and in that case would the cyclical dependency occur. So let's find out some common solution together 👍

Comment (juriczech) [reply suite-native/navigation/src/components/TabBar.tsx:45]: Cyclical dependency won't be increased. Navigation is now used in almost all modules same as wallet-core. It's designed to sustain this scenario right now. 

I believe it shouldn't be controlled by the function you sent, seems too hacky IMO and this solution is much cleaner and deterministic.

Comment (Nodonisko) [reply suite-native/navigation/src/components/TabBar.tsx:45]: I would say it's fine to have it there, from that two added packages is one from common which will never import anything from native and second is feature-flags which should never use navigation.

Also module-home is not only place that is responsible for TabBar because every item in TabBar is basically module so all of them have same rights to mess with it.

I would say let's keep it there because in the end it's the simplest solution without using any "hacks".

Comment (PeKne) [reply suite-native/navigation/src/components/TabBar.tsx:45]: OK, but I will still need to find some "hack" for hiding and showing tab bar based on the `@suite-native/receive` module. 

But I don't want to block this issue anymore with this. ✅

**Review by PeKne — APPROVED** [review]

File: suite-native/navigation/src/components/TabBar.tsx (hunk 17 above)
Comment (Nodonisko) [reply suite-native/navigation/src/components/TabBar.tsx:45]: @PeKne You should be able to track which screens are "present" in navigation package and show tabbar based on that

Comment (PeKne) [reply suite-native/navigation/src/components/TabBar.tsx:45]: I was looking for something that hides the tabbar when you are already in the screen and the screen state changes. But after discussion with design, It is not needed anymore.

**Review by vytick — APPROVED** [review]
:shipit:
